File: 07/main.py
def get_spent_fuel(i, data):
    fuel = 0

    for d in data:
        if d < i:
            fuel += i - d
        elif d > i:
            fuel += d - i
    return fuel


with open("input.txt") as f:
    data = [int(x) for x in f.read().split(",")]

lowest = (-1, 999999)
for i in range(2000):
    fuel = get_spent_fuel(i, data)
    if fuel < lowest[1]:
        lowest = (i, fuel)

print(lowest)

File: 07/main.cpp
#include <fstream>
#include <iostream>
#include <vector>

#include "../common.h"

using namespace std;

int get_spent_fuel(int target, vector<int> data)
{
    int fuel = 0;

    for (int i = 0; i < data.size(); i++) {
        if (data[i] < target) {
            for (int j = 1; j <= target - data[i]; j++) {
                fuel += j;
            }
        } else if (data[i] > target) {
            for (int j = 1; j <= data[i] - target; j++) {
                fuel += j;
            }
        }
    }

    return fuel;
}

int main()
{
    ifstream fp("input.txt");
    vector<int> data = read_ints_comma(fp);

    cout << mean(data) << "\n";
    cout << get_spent_fuel(mean(data), data) << "\n";

    return 0;
}